utils/id.go

```diff
@@ -0,0 +1,19 @@
+package utils
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+// GenerateID generates an ID for a resource in Mongo
+func GenerateID() string {
```

Comment:
This being in the `utils` folder will effect our unit test coverage. Either add a unit test for it or move it to the `mongo.go` file as this has exemption from code coverage. It's a simple enough function and specific enough to how practitioners are stored in mongo that it could be moved to `mongo.go` I think. Either way rename it to something like `GeneratePractitionerId` as this is specific to practitioners and we're unlikely to use it elsewhere.

Reply:
I've ended up adding this method into the transformer which makes the most sense because I can instantly create a self link for the object within the transformer as opposed to doing within the `mongo.go` file. I will create a simple unit test to test this method because it won't exempt from testing if it isn't in the `mongo.go` file. I initially named the method `GeneratePractitionerID()`, but I thought to keep it generic for now as our architect has specified that he would prefer the app to generate IDs rather than using Mongo's objectID.